harness/determined/util.py

```diff
@@ -20,6 +20,11 @@
 
 @util.preserve_random_state
 def download_gcs_blob_with_backoff(blob: Any, n_retries: int = 32, max_backoff: int = 32) -> Any:
+    from google.cloud import storage
+
+    assert isinstance(
```

Comment:
Ok, sorry I didn't think about this on the first time through.

Two things.  First: let's use an `if` here instead of an `assert`.  Python (inexplicably) has a performance mode where asserts don't run.  Since ultimately this is validation results from outside the system (user input of a filename, or gcp server's response) let's avoid `assert` in this case.

Second, while this will fix the 15-minute hang, which is clearly an improvement, can we catch the error somewhere more useful?  I'm thinking if the filepath provided was invalid, we should detect the failure right after `bucket.get_blob()` so that we can return an error with the filename, rather than saying "Called download_gcs_blob_with_backoff with object of type None", which would be rather opaque for the user.

Also, I'm not opposed to leaving the None check here too, as a safeguard.

Reply:
Was not familiar with no-asserts mode!  Will switch to an if -> raise.

Agree on catching the error somewhere more useful as well.  All users of `bucket.get_blob` should be checking this -- but as I ran into this on two separate occasions (one in my code, one in someone else's) it's clear that often doesn't happen.  I think users generally expect that get_blob will raise an exception if it fails, or return an object that will later cause an exception to be raised.  I also have it on my TODO to fix both locations I'm aware of -- this was just a quicker fix to test.